### ISSUE TITLE
Zoom to bounds not working properly when map is tilt or rotated

### ISSUE THREAD

**Issue by johanneslaubermoovel** [mapbox/mapbox-gl-native#10133]
In the video you see when the map is not tilt, zoom to bounds works. But after tilting it, it zooms somewhere else, only after pressing zoom to bounds a second time it works correctly again.

https://youtu.be/v8jU3GUqQtU

```
zoomto.setOnClickListener {
            val latLng1 = LatLng(48.764183, 9.1668824)
            val latLng2 = LatLng(48.7648268, 9.1751567)
            val latLng3 = LatLng(48.7696245, 9.1717453)

            val latLngs = LatLngBounds.Builder()
                    .include(latLng1)
                    .include(latLng2)
                    .include(latLng3)


            val polylineOptions = PolylineOptions().apply {
                color(Color.GREEN)
                width(10f)
                add(latLng1)
                add(latLng2)
                add(latLng3)
            }

            (activity as MainActivity).map_view.getMapAsync { map ->
                map.animateCamera(CameraUpdateFactory.newLatLngBounds(latLngs.build(), 0))
                map.addPolyline(polylineOptions)
            }
        }
```

**Platform:** android
**Mapbox SDK version:** 5.1.3

### Steps to trigger behavior

 1. tilt the map
 2. zoom to bounds

### Expected behavior
Zooms to bounds

### Actual behavior
Zooms somewhere else

**Comment by asheemmamoowala** [comment]
Duplicate of #2259.
